jsmiddleware/src/main/java/com/segment/jsmiddleware/EdgeFunctionMiddlewareProcessor.kt

```diff
@@ -2,80 +2,48 @@ package com.segment.jsmiddleware
 
 import android.content.Context
 import android.util.Log
-import okhttp3.*
-import org.apache.commons.io.IOUtils
-import org.json.JSONObject
+import com.segment.analytics.internal.Utils as SegmentUtils
 import java.io.*
-import java.nio.charset.StandardCharsets
-import kotlin.jvm.Throws
 
-class EdgeFunctionMiddlewareProcessor(private val writeKey: String, private val listener: EdgeFunctionMiddlewareProcessorListener) {
-    private val client = OkHttpClient()
+class EdgeFunctionMiddlewareProcessor(private val listener: EdgeFunctionMiddlewareProcessorListener) {
 
-    ////
-    // Public Methods
     @Throws(Exception::class)
-    fun configureLocalFile(mobContext: Context, localFile: String) {
+    fun configureCachedFile(mobContext: Context) {
+        listener.jsRetrieved(getCachedFileContents(mobContext))
+    }
+
+    @Throws(Exception::class)
+    fun configureFallbackFile(mobContext: Context, localFile: String) {
         val localJSMiddlewareInputStream = mobContext.assets.open(localFile)
-        val result = IOUtils.toString(localJSMiddlewareInputStream, StandardCharsets.UTF_8)
+        val result = readInputStream(localJSMiddlewareInputStream)
         listener.jsRetrieved(result)
     }
 
     @Throws(Exception::class)
-    fun downloadFileAsync(mobContext: Context, downloadUrl: String?) {
-
-        // Grab settings first
-//        try {
-//            val settings = fetchSettings()
-//        } catch (e: Exception) {
-//            Log.e("JSMiddleware", "Could not parse settings")
-//        }
-        val request = Request.Builder().url(downloadUrl!!).build()
-        client.newCall(request).enqueue(object : Callback {
-            override fun onFailure(call: Call, e: IOException) {
-                e.printStackTrace()
-            }
-
-            @Throws(IOException::class)
-            override fun onResponse(call: Call, response: Response) {
-                if (!response.isSuccessful) {
-                    throw IOException("Failed to download file: $response")
-                }
-                try {
-                    response.body.use { responseBody ->
+    fun downloadFile(mobContext: Context, downloadUrl: String) {
+        var connection: ConnectionUtils.Connection? = null
+        try {
+            connection = downloadEdgeFunctionBundle(downloadUrl)
+            if (connection.inputStream != null) {
+                val inputStream = connection.inputStream!!
```

Comment:
May want to consider `connection.inputStream?.let { }` here instead

Reply:
ah yes! good catch